### PR TITLE
Handle responses that are errors

### PR DESCRIPTION
**Pull request by Crazyglue**

### CODE REVIEW

File: ui/frontend/src/components/ActivityStats.vue [hunk 1]
```diff
@@ -19,7 +20,7 @@
         <h3>Summary:</h3>
 
         <ul>
-          <li v-for="activity in activities">
+          <li v-for="(activity, index) in activities" :key="index">
```
Comment (jameslamb) [inline ui/frontend/src/components/ActivityStats.vue]: teach me! what does `:key="index"` do?

File: ui/frontend/src/components/ActivityStats.vue [hunk 2]
```diff
@@ -52,18 +51,29 @@ export default {
       user: "",
       activities: [],
       github_user: "",
+      error: null
+    }
+  },
+  computed: {
+    errorText() {
+      const { status, statusText } = this.error
```
Comment (jameslamb) [inline ui/frontend/src/components/ActivityStats.vue]: can you change `status` to `status_code` (assuming that's what this is?)

File: ui/frontend/src/components/ActivityStats.vue [hunk 3]
```diff
@@ -52,18 +51,29 @@ export default {
...
+      return `${status} -- ${statusText}`;
     }
   },
   methods: {
     // get events for a user
-    get_events: function(){
+    get_events() {
```
Comment (jameslamb) [inline ui/frontend/src/components/ActivityStats.vue]: nice, nice

File: ui/frontend/src/components/ActivityStats.vue [hunk 4]
```diff
@@ -52,18 +51,29 @@ export default {
...
-      .then(function(data) {
+      .then(data => {
         var parsed = JSON.parse(data.bodyText);
         this.events = parsed.events;
         this.user = parsed.user;
         this.num_events = parsed.total;
         this.activities = parsed.activities;
+        this.error = null;
+      })
+      .catch(err => {
```
Comment (jameslamb) [inline ui/frontend/src/components/ActivityStats.vue]: TIL you can add a `.catch` here. This is nice!

File: ui/frontend/src/components/ActivityStats.vue (hunk 1 above)
Comment (Crazyglue) [reply ui/frontend/src/components/ActivityStats.vue]: When you do a `v-for`, if you don't provide a `key`, Vue doesnt know which element corresponds to which array item. So Vue has to perform a re-render on all elements even if only 1 element in the array was mutated. This creates that mapping so Vue only re-renders the element that changed 💨

File: ui/frontend/src/components/ActivityStats.vue (hunk 2 above)
Comment (Crazyglue) [reply ui/frontend/src/components/ActivityStats.vue]: Done :tada

File: ui/frontend/src/components/ActivityStats.vue (hunk 1 above)
Comment (jameslamb) [reply ui/frontend/src/components/ActivityStats.vue]: cool!

**Pull request merged**